examples/beamforming/mvdr.py

```diff
@@ -0,0 +1,418 @@
+            (Default: ``False``)
+    """
+
+    def __init__(
+        self,
+        ref_channel: int = 0,
+        solution: Optional[str] = "ref_channel",
+        diag_loading: bool = True,
+        diag_eps: float = 1e-7,
+        online: bool = False,
+    ):
+        super().__init__()
+        assert solution in ["ref_channel", "stv_evd", "stv_power"],\
+            "Unknown solution provided. Must be one of [``ref_channel``, ``stv_evd``, ``stv_power``]."
+        self.ref_channel = ref_channel
+        self.solution = solution
+        self.diag_loading = diag_loading
+        self.diag_eps = diag_eps
+        self.online = online
+        self.psd = torch.jit.trace(PSD(), (torch.rand(2, 129, 6, 100, dtype=torch.cdouble), torch.rand(2, 129, 100)))
+
+        psd_s: Optional[torch.Tensor] = torch.zeros(1)
+        psd_n: Optional[torch.Tensor] = torch.zeros(1)
+        mask_sum_s: Optional[torch.Tensor] = torch.zeros(1)
+        mask_sum_n: Optional[torch.Tensor] = torch.zeros(1)
+        self.register_buffer('psd_s', psd_s)
+        self.register_buffer('psd_n', psd_n)
+        self.register_buffer('mask_sum_s', mask_sum_s)
+        self.register_buffer('mask_sum_n', mask_sum_n)
+
+    def update_mvdr_vector(
+        self,
+        psd_s: torch.Tensor,
+        psd_n: torch.Tensor,
+        mask_s: torch.Tensor,
+        mask_n: torch.Tensor,
+        reference_vector: torch.Tensor,
+        solution: str = 'ref_channel',
+        diagonal_loading: bool = True,
+        diag_eps: float = 1e-7,
+        eps: float = 1e-8,
+    ) -> torch.Tensor:
+        r"""Recursively update the MVDR beamforming vector.
+
+        Args:
+            psd_s (torch.Tensor): psd matrix of target speech
+            psd_n (torch.Tensor): psd matrix of noise
+            mask_s (torch.Tensor): T-F mask of target speech
+            mask_n (torch.Tensor): T-F mask of noise
+            reference_vector (torch.Tensor): one-hot reference channel matrix
+            solution (str): the solution to estimate the beamforming weight
+                (Default: ``ref_channel``)
+            diagonal_loading (bool): whether to apply diagonal loading to psd_n
+                (Default: ``True``)
+            diag_eps (float): The coefficient multipied to the identity matrix for diagonal loading
+                (Default: 1e-7)
+            eps (float): a value added to the denominator in mask normalization. (Default: 1e-8)
+
+        Returns:
+            torch.Tensor: the mvdr beamforming weight matrix
+        """
+        if self.psd_s == 0:
+            self.psd_s = psd_s
+            self.psd_n = psd_n
+            self.mask_sum_s = mask_s.sum(dim=-1)
+            self.mask_sum_n = mask_n.sum(dim=-1)
+            return self.get_mvdr_vector(psd_s, psd_n, reference_vector, solution, diagonal_loading, diag_eps, eps)
+        else:
+
+            psd_s = self.update_psd_speech(psd_s, mask_s)
```

Comment:
thoughts on using a function name like `get_updated_psd_speech` or having the function directly modify `self.psd_s`? The current function name makes it seem as if updating is part of the function